packages/wallet-ui/src/hooks/useHasMetamask.ts

```diff
@@ -36,13 +46,20 @@ export const useHasMetamask = () => {
   };
 };
 
-export const detectMetamask = async () => {
+export const detectMetamask = async (): Promise<EIP6963ProviderDetail | null> => {
+  return new Promise((resolve) => {
+    const handleProvider = (providerDetail: any) => {
```

Comment:
try not to use any

handleProvider: (providerDetail: EIP6963ProviderDetail) => HandlerReturnType